process/transactionProcessor.go

```diff
@@ -738,3 +743,277 @@ func (tp *TransactionProcessor) getNodesInShard(shardID uint32, reqType requestT
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return observers, sndShardID, nil
+}
+
+func (tp *TransactionProcessor) getTxPool(fields string) (*data.TransactionsPool, error) {
+	shardIDs := tp.proc.GetShardIDs()
+	txs := &data.TransactionsPool{
+		RegularTransactions:  make([]data.WrappedTransaction, 0),
+		SmartContractResults: make([]data.WrappedTransaction, 0),
+		Rewards:              make([]data.WrappedTransaction, 0),
+	}
+	for _, shard := range shardIDs {
+		intraShardTxs, err := tp.getTxPoolForShard(shard, fields)
+		if err != nil {
+			continue
+		}
+
+		txs.RegularTransactions = append(txs.RegularTransactions, intraShardTxs.RegularTransactions...)
+		txs.Rewards = append(txs.Rewards, intraShardTxs.Rewards...)
+		txs.SmartContractResults = append(txs.SmartContractResults, intraShardTxs.SmartContractResults...)
+	}
+
+	hasTxs := (len(txs.RegularTransactions) + len(txs.Rewards) + len(txs.RegularTransactions)) > 0
+	if !hasTxs {
+		return nil, errors.ErrTransactionsNotFoundInPool
+	}
+
+	return txs, nil
+}
+
+func (tp *TransactionProcessor) getTxPoolForShard(shardID uint32, fields string) (*data.TransactionsPool, error) {
+	observers, err := tp.getNodesInShard(shardID, requestTypeObservers)
+	if err != nil {
+		log.Trace("cannot get observers for shard", "shard", shardID, "error", err)
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		txs, ok := tp.getTxPoolFromObserver(observer, fields)
+		if !ok {
+			continue
+		}
+
+		hasTxs := (len(txs.RegularTransactions) + len(txs.Rewards) + len(txs.RegularTransactions)) > 0
+		if !hasTxs {
+			continue
+		}
+
+		return txs, nil
+	}
+
+	log.Trace("cannot get tx pool for shard", "shard", shardID, "error", errors.ErrTransactionsNotFoundInPool.Error())
+	return nil, errors.ErrTransactionsNotFoundInPool
+}
+
+func (tp *TransactionProcessor) getTxPoolFromObserver(
+	observer *data.NodeData,
+	fields string,
+) (*data.TransactionsPool, bool) {
+	txsPoolResponse := &data.TransactionsPoolApiResponse{}
+	apiPath := TransactionsPoolPath + fieldsParam + fields
+
+	respCode, err := tp.proc.CallGetRestEndPoint(observer.Address, apiPath, txsPoolResponse)
+	if err != nil {
+		log.Trace("cannot get tx pool", "address", observer.Address, "error", err)
+
+		if respCode == http.StatusTooManyRequests {
+			log.Warn("too many requests while getting tx pool", "address", observer.Address)
+		}
+
+		return nil, false
+	}
+
+	if respCode != http.StatusOK {
+		return nil, false
+	}
+
+	return &txsPoolResponse.Data.Transactions, true
+}
+
+func (tp *TransactionProcessor) getTxPoolForSender(sender, fields string) (*data.TransactionsPoolForSender, error) {
+	observers, _, err := tp.getShardObserversForSender(sender, requestTypeObservers)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		txsForSender, ok := tp.getTxPoolForSenderFromObserver(observer, sender, fields)
+		if !ok {
+			continue
+		}
+
+		hasTxs := len(txsForSender.Transactions) > 0
```

Comment:
shouldn't be an error so no need to continue

Reply:
done